datafusion/core/src/datasource/listing/table.rs

```diff
@@ -773,10 +788,9 @@ mod tests {
         ];
 
         for (file_sort_order, expected_result) in cases {
-            let options = ListingOptions {
-                file_sort_order,
-                ..options.clone()
-            };
+            let options = ListingOptions::from_options(options.clone())
+                .with_file_sort_order(file_sort_order);
```

Comment:
Would this work? I think we can avoid adding `from_options`

```suggestion
            let options = options.clone()
                .with_file_sort_order(file_sort_order);
```